src/main/java/com/artipie/http/SliceByPath.java

```diff
@@ -98,4 +93,25 @@ public Response response(final String line, final Iterable<Map.Entry<String, Str
             );
         }
     }
+
+    /**
+     * Key from path.
+     * @param path Path from request line
+     * @return Key from path.
+     * @throws IOException In case of problems with reading settings.
+     */
+    private Key keyFromPath(final String path) throws IOException {
+        final String[] split = path.replaceAll("^/+", "").split("/");
+        Key key = Key.ROOT;
+        if (this.settings.layout().equals("org")) {
+            if (split.length >= 2) {
+                key = new Key.From(split[0], split[1]);
+            }
+        } else {
+            if (split.length >= 1) {
+                key = new Key.From(split[0]);
+            }
+        }
+        return key;
```

Comment:
@genryxy using not `final` variables should be avoided. Mutable values make code harder to understand. Please change `key` variable to be `final`.